src/sage/rings/polynomial/polynomial_element.pyx

```diff
@@ -2151,6 +2151,18 @@ cdef class Polynomial(CommutativePolynomial):
             True
             sage: f % factor == 0
             True
+
+        TESTS:
+
+        Ensure that :issue:`37445` is fixed::
+
+            sage: R.<x> = GF(13)[]
+            sage: def irr(d, R): return f.monic() if (f := R.random_element(d)).is_irreducible() else irr(d, R)
```

Comment:
Beautiful code! Great use of recursion! Nice syntax!

Reply:
lol